client/internal/wsreceiver_test.go

```diff
@@ -87,7 +88,7 @@ func TestServerToAgentCommand(t *testing.T) {
 			}
 			sender := WSSender{}
```

Comment:
Can you add the test here also?

Reply:
Yes, added in 40738666b7b4b49ad61baa485c0901cff01c1982. The logic is the same, the test would fail via the race detector if there's no sufficient protection